lib/smart_answer_flows/energy-grants-calculator.rb

```diff
@@ -470,6 +470,9 @@ def define
         precalculate :flat_type do
           flat_type
         end
+        precalculate :under_green_deal do
```

Comment:
@ikennaokpala: I agree about being more readable. Unfortunately this isn't possible when using `precalculate` or other similar blocks. This is because the result is assigned to the state using an attribute writer derived from the block's key. In this case that would mean `state.under_green_deal? = true` which gives a `SyntaxError`. We could look at fixing this, but I think the better solution is to extract a calculator/policy object as per the [refactoring docs][1].

[1]: https://github.com/alphagov/smart-answers/blob/master/doc/refactoring.md